### ISSUE TITLE
No Support for the Nacon Revolution pro controller ??

### ISSUE THREAD

**Issue by Skallas97** [Jays2Kings/DS4Windows#275]
I just go the Nacon Revolution pro ps4 controller but it does not get picked up in the ds4 windows app please help it was really expensive and i cant return it

**Comment by Ryochan7** [comment]
Hopefully the only problem is a different VID + PID combination being used by the device that DS4Windows is not looking for. Is the Nacon Revolution detected by Windows as a game controller? Can you access the controller properties in Device Manager and check the VID and PID, and maybe REV, of the HID-compliant game controller?

**Comment by Skallas97** [comment]
Hi thanks for the quick reply , the controller is picked up by windows as a
game controller i found the information you were looking for and i added a
snip of it as an attachment to this email

**Comment by Ryochan7** [comment]
It didn't go through. It would probably be better to share that using Dropbox or Google Drive.

**Comment by Skallas97** [comment]
https://drive.google.com/file/d/0B-MLxB4feTfxd2FGMWo4NHVIb1ZpNHR2YlpBb1JHLW5kWmhV/view?usp=drivesdk

**Comment by Ryochan7** [comment]
Neither the VID nor PID were included in the device search routine. I have changed DS4Windows to include the VID and PID of the Nacon Revolution controller in its search. As long as the HID reports are the same as a standard PS4 controller then it should work.

https://dl.dropboxusercontent.com/u/31073509/DS4Windows_Test_20170311.zip

**Comment by Skallas97** [comment]
I tried the new Application however the controller is still not being
recognized by DS4Windows is there anyway i could check the HID reports and
give you the information ?

**Comment by Ryochan7** [comment]
Does DS4Windows show anything in the Log tab regarding detecting the controller at all? Even though the code is structured so the device might attempt to read and then remove the device before a log message would appear, there is a chance that DS4Windows would acknowledge that the device is detected before failing to read a report.

I don't know how to obtain the HID report descriptors for a device on Windows; obtaining an HID report descriptor looks like an easier task to perform on Linux. Even with the HID report descriptors, it might be too much work to modify DS4Windows to get it to work with the Nacon Revolution. I was just hoping that the Nacon Revolution would still conform to standard PS4 controller reports so the code would only have to be changed to check for the device.

**Comment by Skallas97** [comment]
I understand i found a way to make the controller work using steams generic
controller support it just isn't as nice as using the ds4windows
application, thank you for your help though ill just have to start playing
ps4 again lol.

**Comment by SiliconExarch** [comment]
I have had limited success in getting my Nacon Revolution to work with DS4Windows. Modifying the search routine allows it to detect the controller however it fails to enumerate it with the error "Index was outside the bounds of the array" unless the HID-compliant vendor-defined device (VID_146B&PID_0D01&REV_0100&MI_03&Col02) is disabled in device manager.

With this device disabled it can enumerate the controller (Exclusive mode is sporadic) and everything apart from the touchpad works (including the gyros and light strip), although the d-pad is stuck scrolling to the left despite no input being shown in the controller properties. Some games tend to ignore this unless you're trying to use the d-pad whereas it renders others unplayable.

It reports a MAC address of all zeroes in the log (makes sense since it doesn't have bluetooth) and often doesn't appear in the Controller list at all when connected aside from the USB icon.

If I can provide any more information to help iron out these issues I'm more than happy to.

**Comment by SiliconExarch** [comment]
I've managed to resolve the issue with the d-pad and have an updated fork here:
https://github.com/Yuki-nyan/DS4Windows

Everything is now working aside from the touchpad and you no longer need to disable any devices. Would anyone mind giving my code a sanity check? I'm far from a confident programmer and I don't want to submit a pull request if there are any glaring errors.

As far as the touchpad goes, I can see the values changing when I use the debug feature to dump the HID data so I can probably fix it although I'm not sure if I'll be able to do so without affecting compatibility with normal DS4 controllers - my current fork works with either controller.

**Comment by Waxo** [comment]
Nice fork Yuki-nyan (i didn't get through the code i'm not a C# developper) but it make my controller works. But i'm still thinking about sending it back. It depends on the nacon response and if DS4 the your fork in the official release.

**Comment by Seletim** [comment]
Thank you Yuki-nyan, very nice work. Like you said, controller works except for the touchpad. But I still had to disable the device in Device-Manager.

**Comment by SiliconExarch** [comment]
I've managed to get the touchpad working now in a separate branch based off Ryochan7's fork here: https://github.com/Yuki-nyan/DS4Windows/releases/tag/v1.4.67-touchpad

All functions should work, both for original DS4 controllers and the Nacon.

**Comment by soyuka** [comment]
@Yuki-nyan great work there, controller does work fine! I have disconnection issues though but they may have been fixed in @Ryochan7 fork. Maybe you could propose your patch there?

**Comment by SiliconExarch** [comment]
I have a release based on a somewhat recent version of his fork, if you've
tried that and it doesn't work it might be the cable. It seems very weak
near to the ferrite bead at the controller end, I've already had one
replacement after mine stopped working entirely and the new one has
developed an intermittent issue within days.

Let me know how you get on, I might update my version of his fork later on.

**Comment by Jax03** [comment]
Hey guys, hope some one can help,i have the exact same issue, Windows 10 picks up the controller steam and the calibration tools to except the mouse-pad, just DS4 and none of the games picks up the controller, anyone aware of a fix soon? This is the event i get.

Device USB\VID_146B&PID_0D01&MI_03\7&442eed&0&0003 could not be migrated.

Last Device Instance Id: USB\VID_046D&PID_C52F&MI_01\7&26fd2c6c&0&0001
Class Guid: {745a17a0-74d3-11d0-b6fe-00a0c90f57da}
Location Path: 
Migration Rank: 0xF000FFFF0000F120
Present: false
Status: 0xC0000719

Thanks in advance.

**Comment by soyuka** [comment]
@Yuki-nyan I merged your patches with @Ryochan7 recent fixes (https://github.com/Ryochan7/DS4Windows). Everything works great! 

here is my fork: https://github.com/soyuka/DS4Windows

**Comment by Jon93c** [comment]
I'm confused how do I use your branches in ds4? Or in general how do I use the solutions you guys came up with? I downloaded the files and then what? 

any help is welcome same issue as the guy who started the post

**Comment by soyuka** [comment]
@Jon93c use https://github.com/Yuki-nyan/DS4Windows/releases/ or open my fork in Visual Studio and find the "build" menu. After building the exe is available under bin/

**Comment by Squirrel1693** [comment]
> @Jon93c use https://github.com/Yuki-nyan/DS4Windows/releases/

Tried this but get the error message "Index was outside the bounds of the array", am I doing it wrong?

**Comment by soyuka** [comment]
About this, Steam does support the Nacon Revolution pro controller!

**Comment by Squirrel1693** [comment]
How can I get it to work on Steam? Doesn't work for me

**Comment by Squirrel1693** [comment]
> How can I get it to work on Steam? Doesn't work for me

Got it to work, apparently you first have to enable controller support in the options. Who knew.

**Comment by soyuka** [comment]
Yeah :) ! Was happy when I found this too. Also it's working without issues for me so far.

**Comment by DragonEMPerror** [comment]
Hey Guys, can somebody please explain to me what I have to do to get the nacon revolution working, step by step. I downloaded ds4 with the zip files, but it doesn't recognize my nacon I need help please.

**Comment by SwizZzDizZz** [comment]
please someone help me ds4 doesent reconise my controller i wont even reconise thr normal ps4 controler what do i do

**Comment by necco1070** [comment]
@DragonEMPerror  did you found a methode to get your controler work ?

**Comment by EminERIKSEN** [comment]
Hi there,i am using a ps4 controller on my PC.It gets connected but when i run DS4 Windows it gives this error 'INDEX WAS OUTSIDE THE BOUNDS OF THE ARRAY'.Please help.Thanks.

**Comment by promailson** [comment]
Windows recognizes a microphone in the control. In order to function I had to disable it. It worked perfectly there.

https://www.flickr.com/photos/140320798@N05/26558998948

**Comment by SativaPhoenix** [comment]
@promailson  Thank you so much, I will try this tonight :) 
Didn't think it would recognize a headphone jack as a microphone, but will check tonight;

Im using the wired controller. LOVE IT :) but DS4windows doesn't find it - trying everything possible

**Comment by promailson** [comment]
@SativaPhoenix If it does not, disable the headset on the nacon as well. What version of your nacon? Mine is version 1.

**Comment by SativaPhoenix** [comment]
Thank you,

I  got this one yesterday:
https://www.takealot.com/nacon-red-wired-compact-controller-ps4/PLID48674042

So I must just check the Windows Microphone/Sound devices? and disable it
there? :) will look in two hours;

**Comment by SativaPhoenix** [comment]
I sent this to the Nacon team:

Question :
Le 14/08/2018 22:03, NACON Gaming (noreply@nacongaming.com) a écrit :
[NACONGAMING] REVOLUTION PRO CONTROLLER - South Africa
From : Darryn Sheldon du Plessis ( darrynphoenix61@gmail.com )
Subject : [NACON GAMING]REVOLUTION PRO CONTROLLER - South Africa

Dear Nacon,
Today, I bought the beautiful PS4 controller _ but ds4windows doesn't
detect it-  and I have to use it as xbox 360 controller. This is fine,
Except that I want to use the touch pad.
Steam thinks it's a ps3 controller.. but it's for PS4; how can we fix
this?: )
Love your products. Thank you. Would love some support.
This e-mail was sent from a contact form on NACON Gaming .

Réponse / Answer :
Greetings,
We have carried the greatest attention to your e-mail.

It is not intended to work on pc.
Only on ps4.

Staying at your service for more information,
The Hot Line Team.


LOL their response was weak..

**Comment by LukeNat0r** [comment]
Did somebody get a fix to this?
I got a Nacon Revolution Pro controller 1

**Comment by promailson** [comment]
If you want to pass me the TeamViewer id I try to help you. My self made it work.

**Comment by LukeNat0r** [comment]
i set up a meeting no remote control

**Comment by TheGunsmith** [comment]
Hi. Unfortunately I do not support the Nacon Revolution Pro controller with the latest DS4. There is something to do. Thanks
 'INDEX WAS OUTSIDE THE BOUNDS OF THE ARRAY

**Comment by soyuka** [comment]
To make it work on steam go to big picture => controller settings => enable ps4 support

**Comment by TheGunsmith** [comment]
Thanks. As soon as I get home I'll try and the above-mentioned audio block. I am writing about the developments

**Comment by promailson** [comment]
Anyone who wants to help, just send me email that I connect via TeamViewer and do the procedures I did in mine to work. Tonight I'm going to go live also doing the procedure to help those who are not getting it. I'll post it later.

**Comment by TheGunsmith** [comment]
> 
> 
> To make it work on steam go to big picture => controller settings => enable ps4 support

Unfortunately, although I've enabled the controller in Steam Big Picture mode, it only works in the game does not work

**Comment by TheGunsmith** [comment]
Köszönöm a közreműködést,  kedves  @promailson !!!!

Probléma megoldva!!!

Teljes támogatás!!!

Nagyon szépen köszönöm!!!

Thanks for your help, dear @promailson !!!!

Problem solved!!!

Full Support !!!

Thank you very much!!!

**Comment by promailson** [comment]
Use this version of DS4 Windows
https://github.com/Ryochan7/DS4Windows/releases

**Comment by Gxspy** [comment]
Hello guys, so ds4windows was working perfectly until I reset my pc.  I installed ds4windows from @Ryochan7 like I usually do and it didn’t work this time, I set it up as a normal install not the program thingy in the left of when you first run ds4windows.  When I plugged in the controller for the first time without ds4windows I heard the windows plugged in sound, but after that when I ran ds4windows it doesn’t work.  I found a pattern as this isn’t the first time it has happened, I’m pretty sure it has to do with shared mode or with the Vig bus or whatever it was called (not a dev or a smart person like you guys) so help would really be apretiated PLEASE, it’s the 3rd week of worldcup in fortnite and I don’t want to miss it

**Comment by mika-n** [comment]
This is wrong forum. This is the obsolete Jays2Kings web site and developers are no longer following this web page. If you are using the newer DS4Windows/Ryochan7 fork then you should jump over to that github web page. Anyway, check your Device Manager and make sure there are no yellow warning icons in any of the devices or any of the HID devices are not in disabled state.

I recommend you to update to V1.7.6 DSWindows version and to update ViGem gamepad device driver also. (or uninstall existing ViGem driver and let DS4Windows to download and install a new version). Few days ago both of those received a new version.

**Comment by UnholyScribe** [comment]
It's 2019 so I don't know if this will even get any attention but my Nacon Revolution pro controller it's not working at all even what the new version of DS4.

**Comment by SiliconExarch** [comment]
> It's 2019 so I don't know if this will even get any attention but my Nacon Revolution pro controller it's not working at all even what the new version of DS4.

I'll take a look. Please can you confirm which version of the Revolution Pro you have and that it still works reliably with a PS4? I found the proprietary cable of the first version to be a point of weakness when testing my original port and had to get a replacement from Nacon.

**Comment by UnholyScribe** [comment]
The Pro Controller 1 and it still works on my PS4. And it works perfectly
fine on Steam games but that's it

**Comment by UnholyScribe** [comment]
Well some Steam games not all of them

**Comment by SiliconExarch** [comment]
> my Nacon Revolution pro controller it's not working at all even what the new version of DS4.

You are using the one from here right: https://github.com/Ryochan7/DS4Windows/releases
I've just tested it with my Pro Controller 1 and everything is working just as I left it c:

**Comment by Edilio1995** [comment]
Hi it's possible to add compatibility to Nacon Wired Compat? 

VID_146B&PID_0603

**Comment by GroundMeat** [comment]
Would it be possible to add Nacon Revolution Unlimited support as well?

VID_146B&PID_0D10

When connected through a cable.

**Comment by JDA-GAMING** [comment]
Bonjour, j'ai une manette nacon wired illuminated compact controller et je n'arrive pas à me connecter sur Ds4 windows. Quelqu'un aurai une solution svp. C'est très important ça fait plusieurs moi que je cherche et tente des trucs mais rien à faire. J'avais acheté cette manette en particulier pour le pad tactile pour l'utiliser à la place de la souris sur les jeux (steam, emulateur etc) merci d'avance

**Comment by mika-n** [comment]
@JDA-GAMING  ?????

Anyway without knowing what you just said then you should post what version of DS4Windows app you are using. Are you using the old V1.4 or V1.5 Jay2Kings version from this github site or are using the newer V2.x Ryochan2 version?  Be aware that if you are using the ancient Jay2Kings version then it hasn't been supported for years and doesn't receive any updates anymore. Ryochan7 version is still supported.
https://github.com/Ryochan7/DS4Windows/releases

**Comment by JDA-GAMING** [comment]
I use version 2.0.10 of Ryochan7 and I cannot connect my controller

**Comment by mika-n** [comment]
@JDA-GAMING  In that case you should post the question to Ryochan7 github forum: https://github.com/Ryochan7/DS4Windows/issues

And while you are posting the question there then attach the content of %APPDATA%\DS4Windows\Logs logfile (or content of Logs tabpage in DS4Windows app in text format) to the issue post. Developers cannot do much without details about the environment, app version and log files. And optionally run the debug version of DS4Windows Ryochan7 version to generate more detailed logfile from a gamepad connection problems.
https://github.com/Ryochan7/DS4Windows/wiki/Troubleshooting#detailed-debug-log-file-from-gamepad-connection-problems

Please don't reply to this post, instead post a question to correct forum (ie. Ryochan7 github forum).

**Comment by RevengeHisoka** [comment]
Bonjour, je n'arrive toujours pas à connecter ma nacon pro evolution controller v1 , pouvez vous m'aidez s'il vous plait , mon pc reconnait la manette , l'application Nacon Sofware aussi mais sur DS4Windows il ne l'as trouve pas

**Comment by mika-n** [comment]
@RevengeHisoka You are using the wrong web site to post a question. This is Jays2Kings version of DS4Windows, but it hasn't been updated for several years and the support here is minimal. 

Instead, you download the latest DS4Windows from Ryochan7/DS4Windows web site and post your question to https://github.com/Ryochan7/DS4Windows/issues web site (and read the pinned "Read this before posting a new issue ticket" post and provide the required details about your environment and app versions and logfiles. And try to run the debug version as instructed there).

And I recommend you to use English (or Finnish) in the post, so there is chances to get help from more people around there in Ryochan7/DS4Windows site.
